### DISCUSSION TITLE
Rearward facing guns and targeting

### DISCUSSION THREAD

**Discussion by xX-Dillinger-Xx** [endless-sky/endless-sky#9876] [Ideas]
I am playing around with adding some rearward facing gun ports but I can't find any way to get them to only fire at targets behind me. All guns fire at the same time. Is there something I'm missing or does that capability not exist currently?
My goal was to equip some missiles at the back to fire rearward to cover my escape. Tracking weapons fired from the front don't really work either.

**Comment by bene-dictator** [comment]
It's not possible to only fire guns/missiles from some hardpoints at the moment. The closest you will get is with missile activation selection.

**Comment by Zitchas** [comment]
You can, however, turn on auto-fire in the settings. If I recall correctly that will fire only guns that are actually pointed at enemies and can hit them.

**Reply by xX-Dillinger-Xx** [reply]
Yeah, currently have auto-fire off. I'll have to test that.

**Comment by xX-Dillinger-Xx** [comment]
I used the Sheragi shard fabricator. Being manual fire, it sort of simulated what I was looking for.  It was kinda interesting and definitely gives the battle a different feel. I wonder how hard it would be to have a dedicated rear firing gun port with it's own independent targeting system. I'm thinking mainly for missile etc. Maybe, even dedicated missiles or torpedos strictly for these rear facing tubes.

**Reply by Zitchas** [reply]
Depending on how one goes about it, not terribly hard. If one wants a ship where all of at least one category of weapons are pointed backwards, then that's super easy, just a matter of setting the appropriate data in the ship definition; and then the appropriate primary or secondary weapon fire key will fire all the rearward facing guns. Very easy to setup.

If one wants to have a separate key just for rear-facing things, then we'd probably be looking at a situation where the best option is to add an "alternate-primary" and "alternate-secondary" fire control options (maybe even simply a modifier+ the regular key for each of those), and then the ship definition specifies which hardpoints are controlled by the regular key and which ones are controlled by the alternate. Even better if that was user selectable via the Ship Info panel. Adding an additional keybind isn't super hard, just moderately time consuming. But tying it in to the whole weapon control scheme and splitting things out and whatnot would take more than a bit of work.

**Comment by xX-Dillinger-Xx** [comment]
Thanks for the info. I was just fooling around with an idea. I did end up using the first option and it was good enough for me to imagine what such a feature might look like. To me, it just looks weird for large ships that can't turn to be firing missiles and torpedoes in the wrong direction just to have them turn around to track the target. It would make more sense to have some independently fire from the back. It would make for better accuracy and efficiency. This feature would be mainly for defensive maneuvers on large ships. Submarines have tubes at the back for a reason. 

Maybe, one day I'll start an 'issue'

**Reply by Zitchas** [reply]
Fun fact, When the graphics were updated, the Bactrian graphics *did* get angled gunports specifically to enable it to fire missiles at targets without having to face them. A huge number of people in the community supported/voted for it. It was rejected as making things "too complicated" as new players in human space who had managed to get Bactrians wouldn't be able to understand  how to use them or some such garbage. (despite the fact that "acquired a Bactrian" and "new player" are rarely, if ever, in the same sentence with each other...)

**Reply by Hecter94** [reply]
Let's try to remain civil, please. 

The actual reasoning given for the usage of angled gunports being reserved for later game content is available here: https://github.com/endless-sky/endless-sky/pull/6509#discussion_r795003295, which I see you read and responded to. Still, please give it another read if you need to refamiliarize yourself with the reasoning behind the decision. 

Since then, we've also added a few other ships utilizing angled gunports, so it's not like it's a niche that isn't being filled. Once some of the UI/AI issues with it get worked out, I could see us adding some more where thematically appropriate, though likely outside of human space, for the reasons given previously. 

I ask that you please stay focused on the future and continue to work productively towards improving Endless Sky rather than staying bogged down in issues that were resolved over two years ago at this point. 



@xX-Dillinger-Xx If you'd like to make an issue, feel free to; I took a quick look through our issue list and didn't see one specifically related to angled ports, but do take a look through as well in case I missed any. 

Some points you'd probably want to touch on in the issue:

1. The Ship UI should display gunport angles in some way. For example, this ship:
![image](https://github.com/endless-sky/endless-sky/assets/15916854/20bee588-e235-4e09-a345-4863a99edcfd)
It has a variety of angled gunports, but when looking at this UI, there's no way to tell which are angled and by how much they are angled, so if they're going to become more commonplace, we'd want some way to display it to the player so they can choose what guns they want in angled ports.

2. The AI would need updating to support angled ports, especially if there are angled fixed guns. Perhaps some on-the-fly DPS calculation with the ship AI (and auto-steer) favoring whatever angle allows it to keep its highest DPS guns on target? Though you'd also probably want a setting, perhaps a personality? That makes a ship ignore its angled ports and only fire them when a target flies in front of it. 

3. The player would need a way to choose what gunports they fire from; the best solution might be firing groups, like what Elite: Dangerous and the X-Series do:
![image](https://github.com/endless-sky/endless-sky/assets/15916854/792d2057-367a-43c8-a357-923f69ad4cd7)
You can set weapons to one of four firing groups (A,B,C,D) as primary or secondary weapons and then fire those weapons whenever you have that specific fire group selected. 
However, that also sounds overly complex, so we'd likely want to streamline that system to fit more in line with the action arcade gameplay of Endless Sky.

4. We'd also need to determine what ships would benefit from angled ports, both thematically and mechanically. Considering we're not looking to add them to human ships, what ships would benefit from them and why? Are we adding ports to those ships? Or moving existing ports? In either case, the overall balance of the ship will be affected, whether positively in the case of adding additional ports or negatively in the case of moving existing ones, so bear that in mind as well.

**Reply by xX-Dillinger-Xx** [reply]
I think it best not to open another can of worms. Maybe, after the AI finishes combat training, I'll revisit.
 
As for not including dedicated rear facing ports on human ships, that doesn't really make sense. In human space is where they are needed the most. Most of the secondary weapons are in human space.  Just my opinion for now.
